hsm/m0hsm_api.c

```diff
@@ -1594,12 +1594,14 @@ int m0hsm_create(struct m0_uint128 id, struct m0_obj *obj,
 		RETURN(-ENOMEM);
 
 	/* make the subobject a single-level layout */
-	m0_composite_layer_add(layout, &subobj, hsm_prio(0, tier_idx));
+	rc = m0_composite_layer_add(layout, &subobj, hsm_prio(0, tier_idx));
+	if (rc != 0)
+		RETURN(rc);
```

Comment:
Is it necessary to do some cleanup before error return?
I mean, the things previously done in create_obj(), m0_client_layout_alloc(), etc.

Reply:
Yes, let me fix that also...

Reply:
Done this in another PR - https://github.com/Seagate/cortx-motr/pull/347.